cli/tools/repl.rs

```diff
@@ -638,23 +637,11 @@ async fn read_line_and_poll(
       response_tx.send(result).unwrap();
     }
 
-    // Because an inspector websocket client may choose to connect at anytime when we have an
-    // inspector server we need to keep polling the worker to pick up new connections.
-    // TODO(piscisaureus): the above comment is a red herring; figure out if/why
-    // the event loop isn't woken by a waker when a websocket client connects.
-    let timeout = tokio::time::sleep(tokio::time::Duration::from_millis(100));
-    pin!(timeout);
-
     tokio::select! {
-      result = &mut line => {
```

Comment:
Just removing this timeout, blocks the whole repl

Reply:
It seems to have been working when we tried it out, but I am going to test it more. The readline is on a blocking task (line 630).

Reply:
> Just removing this timeout, blocks the whole repl

Just tested it and that didn't happen. David will do more testing and finish the PR if it works.

Reply:
Sorry, there's a difference between what I tried. 

It seems you also removed `poll_worker` boolean. Now the process spins at 100% CPU

Reply:
Nice and crispy 100% cpu usage.

Reply:
> It seems you also removed `poll_worker` boolean. Now the process spins at 100% CPU

I think this could be avoided by not polling the event loop again if it returns `Poll::Ready(_)`.

Reply:
@piscisaureus yup, Bartek helped me out with that. This PR should be good now!